_posts/2023-10-10-Debezium-Operator-Takes-off-to-the-Clouds.adoc

```diff
@@ -0,0 +1,426 @@
+
+== Preparing the Environment
+
+Before we get to deploying the operator and consequently Debezium Server itself, we first need an environment to deploy into. In this section we will showcase how to provision a local Kubernetes cluster with running PostgreSQL database and Apache Kafka broker. Note that it is not required for the database and/or the Kafka broker to run inside Kubernetes, we just chose to do so for the purpose of this demonstration.
+
+=== Running a Local Kubernetes Cluster
+You can skip this part if you already have a running Kubernetes cluster available, however make sure you posses a cluster-admin permissions as these are required for operator instalaltion. If not, then read on.
+
+There are multiple tools available to run a local k8s cluster, such as https://minikube.sigs.k8s.io/docs/[Minikube], https://kind.sigs.k8s.io/[Kind], or https://docs.docker.com/desktop/kubernetes/[Docker Desktop]. In ths article we will be using Kind to create a local single node cluster.
+
+
+**Prerequisities**
+
+1. https://kubernetes.io/docs/tasks/tools/#kubectl[Install] `kubectl` distribution for your platform
+2. https://kind.sigs.k8s.io/docs/user/quick-start/#installation[Install] `Kind` distribution for your platform. 
+
+Once you have both, `kubectl` and `kind` installed, create a local Kubernetes cluster by executing the following:
+
+[source, bash]
+----
+kind create cluster --name debezium
+----
+
+We can now configure the cluster context for `kubectl` and check the status of our new k8s environment by running the following command:
+
+[source, bash]
+----
+$ kubectl cluster-info --context kind-debezium
+
+Kubernetes control plane is running at https://127.0.0.1:64815
+CoreDNS is running at https://127.0.0.1:64815/api/v1/namespaces/kube-system/services/kube-dns:dns/proxy
+----
+
+=== Deploying the infrastructure
+In this section we will prepare the required infrastructure -- PostgreSQL database as well as an instance of Kafka broker. 
+
+==== PostgreSQL Database 
+Lets start with deploying the PostgreSQL database by executing the command bellow. 
+
+[CAUTION]
+For simplicity we are using an ephemeral volume mounts which means that any modification done to our database instance will not persist when the pod is recreated. 
+
+[source, bash]
+----
+kubect create -f https://raw.githubusercontent.com/debezium/debezium-operator/2.4/examples/postgres/001_postgresql.yml -n debezium
+----
+
+The yaml file fed to `kubectl` contains several kubernetes manifests:
+
+.001_postgresql.yml
+[source, yaml]
+----
+apiVersion: v1 <1>
+kind: Secret
+metadata:
+  name: postgresql-credentials
+type: opaque
+data:
+  POSTGRES_DB: ZGViZXppdW0=
+  POSTGRES_USER: ZGViZXppdW0=
+  POSTGRES_PASSWORD: ZGViZXppdW0=
+---
```

Comment:
@Naros Would you happen to know how to escape this? The syntax highlight is behaving strangely here.

Reply:
Have you tried with `\---` or `+---+`?
More details [here](https://docs.asciidoctor.org/asciidoc/latest/subs/prevent/) and [here](https://docs.asciidoctor.org/asciidoc/latest/pass/pass-macro/)

Reply:
Yes, doesn't seem to work inside code block.